docker-entrypoint.sh

```diff
@@ -126,19 +126,22 @@ if [ "$1" = 'rabbitmq-server' ]; then
 			].
 		EOF
 	fi
+fi
 
-	if [ "$ssl" ]; then
-		# Create combined cert
-		cat "$RABBITMQ_SSL_CERT_FILE" "$RABBITMQ_SSL_KEY_FILE" > /tmp/combined.pem
-		chmod 0400 /tmp/combined.pem
+if [ "$ssl" ]; then
+	# Create combined cert
+	# If calling this script with an exec this is problematic, as it needs to be owned 400 for perms but this code can run as root OR rabbitmq.
+	# As a workaround prefix the combined cert with the username so each user only operates on the file they control.
+	combined_file="/tmp/$(id -un)_combined.pem"
```

Comment:
@yosifkit Excellent point, I do like that solution better.  As I think about it I don't see any reason to operate as root inside the container, I would expect the rabbitmq user to be sufficient for running and managing the instance.

So, if we opt to just use rabbitmq for everything is using gosu even required, or could a simple USER in the Dockerfile be used instead?  I'm not familiar with gosu so I don't know what advantages it might have over USER in the Dockerfile which, to me, is simpler.